wallet/src/main/scala/org/bitcoins/wallet/internal/TransactionProcessing.scala

```diff
@@ -193,47 +226,62 @@ private[wallet] trait TransactionProcessing extends WalletLogger {
   private def processExistingIncomingTxo(
       transaction: Transaction,
       confirmations: Int,
+      blockHashOpt: Option[DoubleSha256DigestBE],
       foundTxo: SpendingInfoDb): Future[SpendingInfoDb] = {
-    if (foundTxo.confirmations < confirmations) {
-      // TODO The assumption here is that double-spends never occur. That's not
-      // the case. This must be fixed when double-spend logic is implemented.
-      logger.debug(
-        s"Increasing confirmation count of txo=${transaction.txIdBE}, old=${foundTxo.confirmations} new=${confirmations}")
-      val updateF =
-        spendingInfoDAO.update(
-          foundTxo.copyWithConfirmations(confirmations = confirmations))
-
-      updateF.foreach(tx =>
-        logger.debug(
-          s"Updated confirmation count=${tx.confirmations} of output=${foundTxo}"))
-      updateF.failed.foreach(err =>
-        logger.error(
-          s"Failed to update confirmation count of transaction=${transaction.txIdBE}",
-          err))
-
-      updateF
-    } else if (foundTxo.confirmations > confirmations) {
-      val msg =
-        List(
-          s"Incoming transaction=${transaction.txIdBE} has fewer confirmations=$confirmations",
-          s"than what we already have registered=${foundTxo.confirmations}! I don't know how",
-          s"to handle this."
+    if (foundTxo.txid != transaction.txIdBE) {
+      val errMsg =
+        Seq(
+          s"Found TXO has txid=${foundTxo.txid}, tx we were given has txid=${transaction.txIdBE}.",
+          "This is either a reorg or a double spent, which is not implemented yet"
         ).mkString(" ")
-      logger.warn(msg)
-      Future.failed(new RuntimeException(msg))
+      logger.error(errMsg)
+      Future.failed(new RuntimeException(errMsg))
     } else {
-      if (foundTxo.txid == transaction.txIdBE) {
-        logger.debug(
-          s"Skipping further processing of transaction=${transaction.txIdBE}, already processed.")
-        Future.successful(foundTxo)
-      } else {
-        val errMsg =
-          Seq(
-            s"Found TXO has txid=${foundTxo.txid}, tx we were given has txid=${transaction.txIdBE}.",
-            "This is either a reorg or a double spent, which is not implemented yet"
-          ).mkString(" ")
-        logger.error(errMsg)
-        Future.failed(new RuntimeException(errMsg))
+      (foundTxo.blockHash, blockHashOpt) match {
+        case (None, Some(blockHash)) =>
+          logger.debug(
```

Comment:
it would be nice if there was some sort of an invariant here that we could put to make sure the `foundTxo` occurs in the `blockHashOpt`.

Reply:
How? We don't have access from `wallet` to the full blockchain. For neutrino mode it's done automatically though, because it receives a full block and `wallet` knows for sure that all transactions occur in that block.